src/H5ESint.c

```diff
@@ -232,6 +249,78 @@ H5ES__create(void)
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5ES__create() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5ES__insert
+ *
+ * Purpose:     Insert a request token into an event set
+ *
+ * Return:      SUCCEED / FAIL
+ *
+ * Programmer:	Quincey Koziol
+ *	        Friday, December 11, 2020
+ *
+ *-------------------------------------------------------------------------
+ */
```

Comment:
@epourmal for some reason, in my mind's eye, I had the impression the new docs-like-code workflow THG started in the PSIP work and has been working towards would result in being able to document functions more like this...

```
static herr_t
 H5ES__insert(
    H5ES_t *es,   /<** the event set object identifier **/
    H5VL_t *connector, /**< the VOL connector identifier **/
    void *request_token, /**< the request token, 
    const char *app_file, /**< the name of the originating application file of the request **/
    const char *app_func, /**< the name of the originating application function of the request **/
    unsigned app_line, /**< the line of application source code of the originating request **/
    const char *caller, /**< the name of the calling function **/
    const char *api_args) /**< the calling function's arguments, separated by commas **/
```

and have that get slurrped into the browseable documentation using whatever automation doc tool workflow you use. Am I totally misunderstanding things?

Also, @gnuoyd, apologies for inserting myself into your conversation here but I am curious...I have read absolutely nothing and have discussed absolutely nothing with anyone about this API and still believe I know what (most of) the arguments to this function are (supposed to be) because of how they are named. I don't know what the difference is between `app_` args and `caller` and `api_args` and I would want further clarification on that. However, I wonder how much more useful the additional documentation (or a version of it that you would prefer) really leads to deeper/better understanding.

Reply:
@markcmiller86, I find the argument names meaningful, too. I object to the 10-line header comment precisely because it takes up so much space while telling us little more about the program than the code itself does. An unadorned imperative sentence will suffice, `/* Insert a request token into an event set. */`.

The current comment is a waste of display space that could hold program meaning, instead.

Reply:
> @epourmal for some reason, in my mind's eye, I had the impression the new docs-like-code workflow THG started in the PSIP work and has been working towards would result in being able to document functions more like this...
> 
> ```
> static herr_t
>  H5ES__insert(
>     H5ES_t *es,   /<** the event set object identifier **/
>     H5VL_t *connector, /**< the VOL connector identifier **/
>     void *request_token, /**< the request token, 
>     const char *app_file, /**< the name of the originating application file of the request **/
>     const char *app_func, /**< the name of the originating application function of the request **/
>     unsigned app_line, /**< the line of application source code of the originating request **/
>     const char *caller, /**< the name of the calling function **/
>     const char *api_args) /**< the calling function's arguments, separated by commas **/
> ```
> 
> and have that get slurped into the browseable documentation using whatever automation doc tool workflow you use. Am I totally misunderstanding things?

Currently, the Doxygen comments are only in the HDF5 public header files.   Adding them to the parameters for internal routines may be useful if the coding standard was updated to push in that direction.  (Although it would be best done incrementally, in the Boy Scout "Leave the code you change in better shape than you found it" way, since it would be a _huge_ job overall :-)